controllers/packageController.js

```diff
@@ -1,43 +1,37 @@
+const Promise = require('bluebird');
+const status = require('http-status');
 const packageService = require('../services/packageService');
 const { sortPackInfoByProgress, sortPackHistoryByDate } = require('../utils');
 
-const getPackage = async (req, res) => {
+exports.getPackage = async (req, res) => {
     try {
         const packInfo = await packageService.getPackageInfo(req.params.package);
 
-        res.status(200).json({ ...packInfo, ...req.requestUser });
+        res.status(status.OK).json({ ...packInfo, ...req.requestUser });
     } catch (error) {
-        res.status(404).json(error);
+        res.status(status.NOT_FOUND).json(error);
     }
 };
 
-const getPackages = async (req, res) => {
+exports.getPackages = async (req, res) => {
     try {
         const packageIds = req.query.id;
-        const promises = [];
 
-        if (!Array.isArray(packageIds)) throw 'must receive list of packages';
+        if (!Array.isArray(packageIds)) throw new Error('must receive list of packages');
 
-        packageIds.forEach(packageId => {
-            promises.push(packageService.getPackageInfo(packageId));
+        const packList = await Promise.map(packageIds, async id => {
+            const pack = await packageService.getPackageInfo(id);
+            return {
+                ...pack,
+                packHistory: pack.packHistory.sort(sortPackHistoryByDate),
+            };
         });
 
-        let packList = (await Promise.all(promises))
-            .map(package => {
-                return {
-                    ...package,
-                    packHistory: package.packHistory.sort(sortPackHistoryByDate),
-                };
-            })
-            .sort(sortPackInfoByProgress);
+        await packList.sort(sortPackInfoByProgress);
 
-        res.status(200).json({ packList, ...req.requestUser });
+        res.status(status.OK).json({ packList, ...req.requestUser });
     } catch (error) {
```

Comment:
you could destruct error message in the catch, instead of creating a variable for it

Reply:
you are right!